src/main/java/io/vertx/pgclient/PostgresConnection.java

```diff
@@ -3,15 +3,20 @@
 import io.vertx.core.AsyncResult;
 import io.vertx.core.Handler;
 
+import java.util.List;
+
 /**
  * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
  */
+
 public interface PostgresConnection {
 
   void exceptionHandler(Handler<Throwable> handler);
 
   void execute(String sql, Handler<AsyncResult<Result>> handler);
 
+  void execute(String sql, List<Object> params, Handler<AsyncResult<Result>> handler);
```

Comment:
I think we should rename it to prepareAndExecute as this is not the same than the simple `execute` and provide overloading to ease API:

- `prepareAndExecute(String sql, Object param1, Handler<AsyncResult<Result>>)`
- `prepareAndExecute(String sql, Object param1, Object param2, Handler<AsyncResult<Result>>)`

up to a few ones (5 or 6)